apps/astarte_realm_management/lib/astarte_realm_management/queries.ex

```diff
@@ -899,255 +813,84 @@ defmodule Astarte.RealmManagement.Queries do
+    keyspace = Realm.keyspace_name(realm_name)
 
     simple_triggers_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(simple_triggers_statement)
-      |> DatabaseQuery.put(:object_id, object_id)
-      |> DatabaseQuery.consistency(:quorum)
+      from SimpleTrigger,
+        where: [object_id: ^object_id, object_type: 2]
 
-    with {:ok, result} <- DatabaseQuery.call(db_client, simple_triggers_query),
-         [count: count] <- DatabaseResult.head(result) do
-      if count != 0 do
-        {:ok, true}
-      else
-        {:ok, false}
-      end
-    else
-      %{acc: _, msg: error_message} ->
-        _ = Logger.warning("Database error: #{error_message}.", tag: "db_error")
-        {:error, :database_error}
-
-      {:error, reason} ->
-        _ = Logger.warning("Failed with reason: #{inspect(reason)}.", tag: "db_error")
-        {:error, :database_error}
-    end
-  end
-
-  def get_jwt_public_key_pem(keyspace) do
-    case Xandra.Cluster.run(:xandra, &do_get_jwt_public_key_pem(keyspace, &1)) do
-      {:ok, pem} ->
-        {:ok, pem}
-
-      {:error, %Xandra.ConnectionError{} = err} ->
-        _ =
-          Logger.warning("Database connection error #{Exception.message(err)}.",
-            tag: "database_connection_error"
-          )
-
-        {:error, :database_connection_error}
-
-      {:error, %Xandra.Error{} = err} ->
-        handle_xandra_error(err)
-    end
-  end
-
-  defp do_get_jwt_public_key_pem(keyspace_name, conn) do
-    query = """
-    SELECT blobAsVarchar(value)
-    FROM #{keyspace_name}.kv_store
-    WHERE group='auth' AND key='jwt_public_key_pem';
-    """
-
-    with {:ok, prepared} <- Xandra.prepare(conn, query),
-         {:ok, page} <-
-           Xandra.execute(conn, prepared, %{},
-             uuid_format: :binary,
-             consistency: :quorum
-           ) do
-      case Enum.to_list(page) do
-        [%{"system.blobasvarchar(value)": pem}] ->
-          {:ok, pem}
-
-        [] ->
-          {:error, :public_key_not_found}
-      end
-    end
+    {:ok, Repo.some?(simple_triggers_query, prefix: keyspace)}
   end
 
-  defp handle_xandra_error(error) do
-    %Xandra.Error{message: message} = error
-
-    case Regex.run(@keyspace_does_not_exist_regex, message) do
-      [_message, keyspace] ->
-        Logger.warning("Keyspace #{keyspace} does not exist.",
-          tag: "realm_not_found"
-        )
-
-        {:error, :realm_not_found}
+  def get_jwt_public_key_pem(realm_name) do
+    keyspace = Realm.keyspace_name(realm_name)
 
-      nil ->
-        _ =
-          Logger.warning(
-            "Database error, cannot get realm public key: #{Exception.message(error)}.",
-            tag: "database_error"
-          )
+    query =
+      from store in KvStore,
+        select: store.value,
+        where: [group: "auth", key: "jwt_public_key_pem"]
 
-        {:error, :database_error}
-    end
+    Repo.fetch_one(query, consistency: :quorum, prefix: keyspace)
```

Comment:
Here the query is no longer selecting `blobAsVarchar(value)`, but just `value`.

To handle the casting logic, you can consider using the same code written by @noaccOS and used in other services for the KvStore, e.g. https://github.com/astarte-platform/astarte/blob/release-1.2/apps/astarte_pairing/lib/astarte_pairing/realms/kv_store.ex

Reply:
Yes, the `KvStore` in realm management ([permalink here](https://github.com/astarte-platform/astarte/blob/1e8147fcc5463d8bfb5916d306eb0858933491c0/apps/astarte_realm_management/lib/astarte_realm_management/astarte/kv_store.ex#L87-L102)) already contains that logic :+1: